src/main/java/com/saralein/server/connection/IServerSocket.java

```diff
@@ -0,0 +1,9 @@
+package com.saralein.server.connection;
+
+import java.io.IOException;
+
+public interface IServerSocket {
```

Comment:
Do you have any thoughts on this naming convention? (`IName` for interfaces)

Reply:
I'm not a huge fan of it.  I did some research on it, but couldn't find a consensus on convention.  Would you name it `ServerSocketInterface` and then the implementation as `ServerSocketImpl`?

Reply:
I like to use the name of the abstraction as the name of the interface, and then use a more specific name to name the implementations.

Java does this for example with its collections. `List` is the interface, and `ArrayList` is an implementation of that. I tend to avoid `IInterface` as well as `Impl`. You have an opportunity to create your own "language" inside of your server -- using `Impl` feels like a cop out to me.

Reply:
Thus, `ServerSocket` is what I would name the interface. I am not sure what I would call the implementation.